app/src/main/java/com/example/cs4520_twitter/data_layer/database/BabDAO.kt

```diff
@@ -0,0 +1,26 @@
+package com.example.cs4520_twitter.data_layer.database
+
+import androidx.room.Dao
+import androidx.room.Insert
+import androidx.room.Query
+
+/**
+ * Bab DAO
+ */
+@Dao
+interface BabDao {
+    @Insert
+    suspend fun insert(bab: BabEntity)
+
+    /**
+     * Get bab given specified bab id
+     */
+    @Query("SELECT * FROM babs WHERE id = :babId")
+    suspend fun getBabById(babId: String): BabEntity?
+
+    /**
+     * Gets all babs associated with given user id
+     */
+    @Query("SELECT * FROM babs WHERE user_id = :userID")
+    suspend fun getAllFromUserByUserID(userID: String)
+}
```

Comment:
Did we also want functionality for deleting a bab? The api call we mentioned in the google doc:  `"/babs/{bab id}"`

Reply:
Yup will add